network/tests/fuzzing.rs

```diff
@@ -137,3 +141,425 @@ async fn fuzzing_post_handshake() {
         node1.write_bytes(&random_bytes).await;
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_corrupted_version_pre_handshake() {
+    // tracing_subscriber::fmt::init();
+
+    let node_setup = TestSetup {
+        consensus_setup: None,
+        ..Default::default()
+    };
+
+    let node = test_node(node_setup).await;
+    let node_addr = node.local_address().unwrap();
+
+    let mut rng = thread_rng();
+
+    for _ in 0..ITERATIONS {
+        let mut stream = TcpStream::connect(node_addr).await.unwrap();
+        let version = Version::serialize(&Version::new(1u64, stream.local_addr().unwrap().port())).unwrap();
+
+        // Replace a random percentage of random bytes at random indices in the serialised message.
+        let corrupted_version: Vec<u8> = version
+            .into_iter()
+            .map(|byte| if rng.gen_bool(0.1) { rng.gen() } else { byte })
+            .collect();
+
+        let header = MessageHeader::from(corrupted_version.len());
+
+        let _ = stream.write_all(&header.as_bytes()).await;
+        let _ = stream.write_all(&corrupted_version).await;
+    }
+
+    assert_eq!(node.peer_book.read().number_of_connected_peers(), 0);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_corrupted_version_post_handshake() {
+    // tracing_subscriber::fmt::init();
+
+    let (node1, mut node2) = spawn_2_fake_nodes().await;
+
+    tokio::spawn(async move {
+        loop {
+            let _ = node2.read_payload().await;
+        }
+    });
+
+    let mut rng = thread_rng();
+
+    for _ in 0..ITERATIONS {
+        let version = Version::serialize(&Version::new(1, 4141)).unwrap();
+
+        // Replace a random percentage of random bytes at random indices in the serialised message.
+        let corrupted_version: Vec<u8> = version
+            .into_iter()
+            .map(|byte| if rng.gen_bool(0.1) { rng.gen() } else { byte })
+            .collect();
+
+        let header = MessageHeader::from(corrupted_version.len());
+
+        node1.write_bytes(&header.as_bytes()).await;
+        node1.write_bytes(&corrupted_version).await;
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_corrupted_empty_payloads_pre_handshake() {
```

Comment:
I understand the the reason for fuzzing messages that are usually immediately rejected in pre-handshake communication is the potential for breakage when they are deserialized (i.e. so it can even be determined that they should be rejected); I'd leave a short comment so that this rationale is clear